sdks/python/apache_beam/utils/shared.py

```diff
@@ -26,22 +26,30 @@
 
 To share a very large list across all threads of each worker in a DoFn::
 
-  class GetNthStringFn(beam.DoFn):
-    def __init__(self, shared_handle):
-      self._shared_handle = shared_handle
-
-    def process(self, element):
-      def initialize_list():
-        # Build the giant initial list.
-        return [str(i) for i in range(1000000)]
-
-      giant_list = self._shared_handle.acquire(initialize_list)
-      yield giant_list[element]
-
-  p = beam.Pipeline()
-  shared_handle = shared.Shared()
-  (p | beam.Create([2, 4, 6, 8])
-     | beam.ParDo(GetNthStringFn(shared_handle)))
+# Several built-in types such as list and dict do not directly support weak
+# references but can add support through subclassing:
+# https://docs.python.org/3/library/weakref.html
+class WeakRefList(list):
+	pass
+
+class GetNthStringFn(beam.DoFn):
+  def __init__(self, shared_handle):
+    self._shared_handle = shared_handle
+
+  def setup(self):
```

Comment:
Be good to have a note why setup is best place to do the init.

Reply:
done